akka-actor/src/main/scala/akka/actor/CoordinatedShutdown.scala

```diff
@@ -435,17 +438,24 @@ final class CoordinatedShutdown private[akka] (
    * begins its shutdown sequence. Added hooks may run in any order
    * concurrently, but they are running before Akka internal shutdown
    * hooks, e.g. those shutting down Artery.
+   *
+   * The hooks are removed if the actor system is terminated.
    */
   @tailrec def addJvmShutdownHook[T](hook: ⇒ T): Unit = {
```

Comment:
If we want to support removal of hooks added by user I think it should be add/remove API, but the user is responsible for calling remove. That would require that the Thread instance is exposed. A better API might be to hide that and return a `Cancellable` here and removal can be performed via that.

That would require change of signature, i.e. new method must be added. Not sure if we have any real world requests for such, or if we should just fix the leak we caused ourselves with `initJvmHook`?